Description: Use --memopt option when calling TOPCOM
 Saves memory using caching to avoid example/test failures.
Author: Doug Torrance <dtorrance@debian.org>
Bug: https://github.com/Macaulay2/M2/issues/1707
Bug: https://github.com/Macaulay2/M2/issues/2712
Last-Update: 2022-12-15

--- a/M2/Macaulay2/packages/Topcom.m2
+++ b/M2/Macaulay2/packages/Topcom.m2
@@ -119,7 +119,7 @@
 topcomIsRegularTriangulation = method(Options=>{Homogenize=>true})
 topcomIsRegularTriangulation(Matrix, List) := Boolean => opts -> (A, tri) -> (
     -- now create the output file
-    (outfile, errfile) := callTopcom("checkregularity --checktriang -v", {topcomPoints(A, opts), [], tri });
+    (outfile, errfile) := callTopcom("checkregularity --checktriang --memopt -v", {topcomPoints(A, opts), [], tri });
     match("[Cc]hecked 1 triangulations, 0 non-regular so far", get errfile)
     )
 
@@ -317,7 +317,7 @@
       << "Index sets do not correspond to full-dimensional simplices" << endl;
       return false;
    );
-   (outfile, errfile) := callTopcom("points2nflips --checktriang -v", {topcomPoints(V, Homogenize=>false), [], T });
+   (outfile, errfile) := callTopcom("points2nflips --checktriang --memopt -v", {topcomPoints(V, Homogenize=>false), [], T });
    not match("not valid", get errfile)
 )
 
